03-Dec-2020 : Login Fix Update

RefaldiErgian opened the pull request:
perbaikan pada loginnya, sekarang sudah bisa login bagi yang sudah verified (didatabase) dan bagi yang tidak

Catatan : kemarin pull requestnya tidak menggunakan akun ini jadi saya pull request lagi skrg

RefaldiErgian committed:
03-Dec-2020 : Login Fix Update

perbaikan pada loginnya, sekarang sudah bisa login bagi yang sudah verified (didatabase) dan bagi yang tidak

Catatan : kemarin pull requestnya tidak menggunakan akun ini jadi saya pull request lagi skrg

ardiansyah-sweng commented on app/Models/Dosen.php:
```diff
@@ -13,4 +13,22 @@ class Dosen extends Model
         'nipy'
     ];
+	
+	public function loginFunctionDosen($id){
```
jangan lupa menuliska komentar fungsinya

ardiansyah-sweng commented on app/Models/Dosen.php:
```diff
@@ -13,4 +13,22 @@ class Dosen extends Model
+	
+	public function loginFunctionDosen($id){
+        if(!isset($id)){
+            echo "hush";
```
echo nya harap diperjelas. "hush" itu maksudnya apakah mengusir? :)

RefaldiErgian replied:
ya pak hush itu buat ngusir, kalau ada yg ngakses modalnya tanpa request atau idnya gk ketemu (yang dimana itu tidak mungkin kecuali di hack)

ardiansyah-sweng commented on app/Models/Dosen.php:
```diff
@@ -13,4 +13,22 @@ class Dosen extends Model
+        $db_decode = json_decode($db_encode, true);
+        $namamhs = $db_decode[0]['nama'];
+        $avatar = $db_decode[0]['avatar'];
+        $array = array($namamhs, $id, true, $avatar);
```
kenapa ada variabel $namamhs? padahal ini fungsi login utk dosen kan?

RefaldiErgian replied:
yang dosen belum jadi pak, masih menggunakan yang lama, dan itu sepertinya salah penamaan variabel, tapi masih bisa ambil nama di tabel dosen

RefaldiErgian committed:
Perbaikan variabel mhs -> dsn + comment

RefaldiErgian reviewed (commented):
Sudah di update

RefaldiErgian commented:
mohon pak di apply, yg punya nanya modelnya masih kosong ( belum di apply )

RefaldiErgian committed:
23-Dec-2020 Update

-Perbaikan pada login dosen
-Perbaikan pada logout
-Pembuatan helper untuk cek login

RefaldiErgian commented:
Update 23-Dec-2020

-Perbaikan pada login dosen
-Perbaikan pada logout
-Pembuatan helper untuk cek login

RefaldiErgian committed:
23-Dec-2020, Update 1.b

Polimorphism

RefaldiErgian commented:
**UNTUK UAS SOAL NO.3**

**Deskripsi**
Seharusnya berjalan dengan baik tapi entah kenapa tidak jalan, array memiliki value yang sama dengan yang di test dengan assert, akan tetapi masih tidak bisa di test

**Langkah-langkah**
1. Buka xampp / aplikasi server lain dan jalankan service apache dan databasenya
2. Jalankan cmd dan change dir ke directory project nya
3. Jalankan command "php artisan test"
4. Tunggu hasilnya

**Screenshots**
Testing :
![image](https://user-images.githubusercontent.com/40123437/102951009-70378200-44fe-11eb-9e04-3f647c3d8a3f.png)

Hasil :
![image](https://user-images.githubusercontent.com/40123437/102950964-585ffe00-44fe-11eb-89df-df9b8fadf534.png)

RefaldiErgian committed:
23-Dec-2020

Update example & 1c
